src/protocols/bdx/TransferFacilitator.cpp

```diff
@@ -42,18 +37,19 @@ CHIP_ERROR TransferFacilitator::OnMessageReceived(chip::Messaging::ExchangeConte
 
     ChipLogDetail(BDX, "%s: message " ChipLogFormatMessageType " protocol " ChipLogFormatProtocolId, __FUNCTION__,
                   payloadHeader.GetMessageType(), ChipLogValueProtocolId(payloadHeader.GetProtocolID()));
-    CHIP_ERROR err =
-        mTransfer.HandleMessageReceived(payloadHeader, std::move(payload), System::SystemClock().GetMonotonicTimestamp());
+    CHIP_ERROR err = mTransfer.HandleMessageReceived(payloadHeader, std::move(payload));
     if (err != CHIP_NO_ERROR)
     {
         ChipLogError(BDX, "failed to handle message: %" CHIP_ERROR_FORMAT, err.Format());
     }
 
-    // Almost every BDX message will follow up with a response on the exchange. Even messages that might signify the end of a
-    // transfer could necessitate a response if they are received at the wrong time.
-    // For this reason, it is left up to the application logic to call ExchangeContext::Close() when it has determined that the
-    // transfer is finished.
-    mExchangeCtx->WillSendMessage();
+    // Almost every BDX message except BlockAckEOF will follow up with a response on the exchange. Even messages that might signify
+    // the end of a transfer could necessitate a response if they are received at the wrong time. For this reason, it is left up to
+    // the application logic to call ExchangeContext::Close() when it has determined that the transfer is finished.
+    if (mExchangeCtx != nullptr && !payloadHeader.HasMessageType(chip::bdx::MessageType::BlockAckEOF))
```

Comment:
So do we send responses to StatusReport messages?

That said, if BlockAckEOF happens at the "wrong" time (e.g. when a block request is expected!), would _that_ not get a StatusResponse?

Reply:
we do not send responses to status report messages. if a blockackeof happens at the wrong time a status report will be sent with appropriate error code. see - https://github.com/project-chip/connectedhomeip/pull/26040/commits/e5213ad18a9342187ec13de8795d643e18197a94#diff-eb27c4ea98e4f9214d05fd68debdcf2291a46ae3d74039b7b7e412aa55ee7708L728